DhsTranslatorTest/src/test/java/edu/gemini/dhs/translator/test/DhsTranslatorTest.java

```diff
@@ -246,6 +258,33 @@ public void testSetKeywordWrongType() throws ClientProtocolException,
         assertEquals("Operation did not fail.", "error", status.asText());
     }
 
+    @Test
+    public void testConcurrentCalls() throws InterruptedException,
+            ExecutionException {
+        final ConnectionRequester connectionRequester = new ConnectionRequester();
+
+        Callable<String> task = new Callable<String>() {
+            @Override
+            public String call() throws Exception {
+                return connectionRequester.requestConnection();
+            }
+        };
+
+        List<Callable<String>> tasks = Collections.nCopies(numberOfThreads,
+                task);
+        ExecutorService executorService = Executors
+                .newFixedThreadPool(numberOfThreads);
+
+        List<Future<String>> futures = executorService.invokeAll(tasks);
+        List<String> resultList = new ArrayList<String>(futures.size());
+
+        for (Future<String> future : futures) {
+            // Throws an exception if an exception was thrown by the task.
+            resultList.add(future.get());
```

Comment:
We might as well use the `get()` variant with a timeout here.  I don't know how long is reasonable to wait but presumably this should be quick if all is well.

Can this test be run automatically though?  Does it require first setting up a server to connect to or is that handled somehow?